stackit/internal/utils/utils_test.go

```diff
@@ -0,0 +1,122 @@
+package utils
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+)
+
+func TestReconcileStrLists(t *testing.T) {
+	tests := []struct {
+		description string
+		list1       []string
+		list2       []string
+		expected    []string
+	}{
+		{
+			"empty lists",
+			[]string{},
+			[]string{},
+			[]string{},
+		},
+		{
+			"list1 empty",
+			[]string{},
+			[]string{"a", "b", "c"},
+			[]string{"a", "b", "c"},
+		},
+		{
+			"list2 empty",
+			[]string{"a", "b", "c"},
+			[]string{},
+			[]string{},
+		},
+		{
+			"no common elements",
+			[]string{"a", "b", "c"},
+			[]string{"d", "e", "f"},
+			[]string{"d", "e", "f"},
+		},
+		{
+			"common elements",
+			[]string{"d", "a", "c"},
+			[]string{"b", "c", "d", "e"},
+			[]string{"d", "c", "b", "e"},
+		},
+		{
+			"common elements with empty string",
+			[]string{"d", "", "c"},
+			[]string{"", "c", "d"},
+			[]string{"d", "", "c"},
+		},
+		{
+			"common elements with duplicates",
+			[]string{"a", "b", "c", "c"},
```

Comment:
did you get to test some resource with duplicates to check if there is any error?

Reply:
Yes, probably depends on API, but DNS allows for repeated items in the request and just ignores the fact that it is repeated.